### PR TITLE
#128 DRep key generation

### PR DESCRIPTION
**Pull request by carbolymer**
# Changelog

```yaml
- description: |
    CLI command 'drep cardano-cli conway governance drep key-gen` for DRep key generation

# uncomment types applicable to the change:
  type:
   - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
   - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Implements #128

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

**Review by disassembler — APPROVED** [review]

File: Setup.hs [hunk 1]
```diff
@@ -1,2 +0,0 @@
-import           Distribution.Simple
```
Comment (Jimbo4350) [inline Setup.hs:1]: Was this intentional?

File: Setup.hs [hunk 2]
```diff
@@ -1,2 +0,0 @@
...
-main = defaultMain
```
Comment (newhoggy) [inline Setup.hs:2]: Is this removed because it isn't needed?

**Comment by newhoggy** [comment]
This looks good and is low risk because it is only adding code.

File: Setup.hs (hunk 2 above)
Comment (carbolymer) [reply Setup.hs:2]: Yes, it's not used when using `cabal-install`

**Review by newhoggy — APPROVED** [review]

File: Setup.hs (hunk 2 above)
Comment (newhoggy) [reply Setup.hs:2]: Nice!

File: cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance.hs [hunk 3]
```diff
@@ -69,3 +73,4 @@ renderEraBasedGovernanceCmds = \case
     renderGovernanceCommitteeCmds cmds
   EraBasedGovernanceActionCmds cmds ->
     renderGovernanceActionCmds cmds
+  EraBasedGovernanceDRepGenerateKey{} -> "governance drep key-gen"
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance.hs:76]: What do you think about having a single key generation command that generates all the different keys? cc: @newhoggy  @disassembler

**Review by Jimbo4350 — APPROVED** [review]
Approving but we should consider my comment: https://github.com/input-output-hk/cardano-cli/pull/163#discussion_r1293446207

File: cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance.hs (hunk 3 above)
Comment (carloslodelar) [reply cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance.hs:76]: Sounds like a really good idea!!

Comment (CarlosLopezDeLara) [reply cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance.hs:76]: sounds like a really good idea!

**Pull request merged**